zerver/lib/actions.py

```diff
@@ -1808,7 +1802,6 @@ def check_typing_notification(sender: UserProfile,
     except ValidationError as e:
         assert isinstance(e.messages[0], str)
         raise JsonableError(e.messages[0])
-    assert recipient.type != Recipient.STREAM
```

Comment:
We'll need to expand the API to allow indicating the `message_type` for the notification somewhere around here and its callers.